go-packages/meep-net-char-mgr/algo-segment.go

```diff
@@ -997,64 +1023,41 @@ func updateMaxFairShareBwPerFlow(segment *SegAlgoSegment) {
 }
 
 // getNetChars - Retrieve all network characteristics from model for provided element name
-func getNetChars(elemName string, model *mod.Model) (nc NetChar) {
+func getNetChars(elemName string, model *mod.Model) (*dataModel.NetworkCharacteristics) {
+
+	nc := new(dataModel.NetworkCharacteristics)
+
 	// Get Node
 	node := model.GetNode(elemName)
 	if node == nil {
 		log.Error("Error finding element: " + elemName)
 		return nc
 	}
 
-	maxThroughput := 0.0
-	latency := 0.0
-	jitter := 0.0
-	packetLoss := 0.0
 	// Get max throughput based on Node Type, as well as other netcharse
 	if p, ok := node.(*dataModel.Process); ok {
-		maxThroughput = float64(p.AppThroughput)
-		latency = float64(p.AppLatency)
-		jitter = float64(p.AppLatencyVariation)
-		packetLoss = float64(p.AppPacketLoss)
+		nc = p.NetChar
 	} else if pl, ok := node.(*dataModel.PhysicalLocation); ok {
-		maxThroughput = float64(pl.LinkThroughput)
-		latency = float64(pl.LinkLatency)
-		jitter = float64(pl.LinkLatencyVariation)
-		packetLoss = float64(pl.LinkPacketLoss)
+		nc = pl.NetChar
 	} else if nl, ok := node.(*dataModel.NetworkLocation); ok {
-		maxThroughput = float64(nl.TerminalLinkThroughput)
-		latency = float64(nl.TerminalLinkLatency)
-		jitter = float64(nl.TerminalLinkLatencyVariation)
-		packetLoss = float64(nl.TerminalLinkPacketLoss)
+		nc = nl.NetChar
 	} else if zone, ok := node.(*dataModel.Zone); ok {
-		if zone.NetChar != nil {
-			maxThroughput = float64(zone.NetChar.Throughput)
-			latency = float64(zone.NetChar.Latency)
-			jitter = float64(zone.NetChar.LatencyVariation)
-			packetLoss = float64(zone.NetChar.PacketLoss)
-		}
+		nc = zone.NetChar
 	} else if domain, ok := node.(*dataModel.Domain); ok {
-		maxThroughput = float64(domain.InterZoneThroughput)
-		latency = float64(domain.InterZoneLatency)
-		jitter = float64(domain.InterZoneLatencyVariation)
-		packetLoss = float64(domain.InterZonePacketLoss)
+		nc = domain.NetChar
 	} else if deployment, ok := node.(*dataModel.Deployment); ok {
-		maxThroughput = float64(deployment.InterDomainThroughput)
-		latency = float64(deployment.InterDomainLatency)
-		jitter = float64(deployment.InterDomainLatencyVariation)
-		packetLoss = float64(deployment.InterDomainPacketLoss)
+		nc = deployment.NetChar
 	} else {
 		log.Error("Error casting element: " + elemName)
 	}
 
 	// For compatiblity reasons, set to default value if 0
-	if maxThroughput == 0 {
-		maxThroughput = DEFAULT_THROUGHPUT_LINK
+	if nc.ThroughputUl == 0 {
+		nc.ThroughputUl = DEFAULT_THROUGHPUT_LINK
```

Comment:
Same comment as before - throughput = 0 will need to be supported.
It is unclear why 0 defaults to 1Mbps

Reply:
same reply as above